File: JS/bmi.js
height = 10
weight = 20


class BmiView {
    bmiController = new BmiController()

    printBMI(bmi) {
        console.log(bmi.value);
    }

    printAdvice(bmi) {
        console.log(bmi.advice);
    }

    printCompleteDetails(bmi) {
        console.log(bmi.value, bmi.advice, bmi.color);
    }

    printBMIdetails(option) {
        let bmi = this.bmiController.calculateBMI(height, weight)
        if (option == 1) {
            this.printBMI(bmi)
        } else if (option == 2) {
            this.printAdvice(bmi)
        } else {
            this.printCompleteDetails(bmi)
        }
    }
}

class BmiController {
    calculateBMI (height, weight) {
        let bmi = new BMI()
        bmi.value =  weight / (height * height)

        bmi.color = "red"
        if (bmi.value < 18.5) {
            bmi.advice = "underweight"
        } else if (bmi.value < 24.9) {
            bmi.advice = "fit"
            bmi.color = "green"
        } else {
            bmi.advice = "overweight"
        }

        return bmi
    }
}

class BMI {
    value;
    advice;
    color;

    constructor (value, advice, color) {
        this.value = value
        this.advice = advice
        this.color = color
    }
}

bmiView = new BmiView()

bmiView.printBMIdetails(2)
bmiView.printBMIdetails(1)
bmiView.printBMIdetails(3)

File: Python/bmi.py

height = 10
weight = 20

class BmiView:
    def __init__(self):
        self.bmiController = BmiController()
        
    def printBMI(self, bmi):
        print(bmi.value)
        
    def printAdvice(self, bmi):
        print(bmi.advice)
    
    def printCompleteDetails(self, bmi):
        print(bmi.value, bmi.advice, bmi.color)
    
    def printBMIdetails(self, option):
        bmi = self.bmiController.calculateBMI(height, weight)
        if option == 1:
            self.printBMI(bmi)
        elif option == 2:
            self.printAdvice(bmi)
        else:
            self.printCompleteDetails(bmi)

class BmiController:
    def calculateBMI(self, height, weight):
        bmi = BMI()
        bmi.value = weight / (height * height)    
        bmi.color = "red"
        if bmi.value < 18.5:
            bmi.advice = "underweight"
        elif bmi.value < 24.9:
            bmi.advice = "fit"
            bmi.color = "green"
        else:
            bmi.advice = "overweight"
        
        return bmi

class BMI:
    def init(self, value=0, advice="", color=""):
        self.value = value
        self.advice = advice
        self.color = color

bmiView = BmiView()

bmiView.printBMIdetails(2)
bmiView.printBMIdetails(1)
bmiView.printBMIdetails(3)